Automattic/woocommerce-payments
# Send translations to GlotPress on release

Fixes #1559 

#### Changes proposed in this Pull Request

This PR adds support for generating a POT translation file and pushing it to GlotPress* including it in the release zip file.

The process consists of 4 steps: 
- A POT files is generated from the strings in the client javascript, using the `babel` plugin `@wordpress/babel-plugin-makepot`
- A second POT files is generated from the strings in the PHP files, using the `node` package `node-wp-i18n`
- Both files get merged using a php script `bin/combine-pot-files.php` (copied from the woocommerce-admin project, but we should convert it to a JS task, maybe needed in this PR?)
- The actual pushing of the POT file is taken care by `woorelease`, all that is needed is to add `"translate": true` to the `package.json` config, and include the POT file in the `languages` folder in the release zip file.

_Note *_: I does not actually send the file to GlotPress, during the release process `woorelease` notifies GlotPress of the new available release and it triggers a job that fetches it and sync the translation strings (this is the reason the POT file needs to be in the release zip file)

_Note 2_: After adding the i18n dependencies to the `package.json` file, some version fixing was needed, the idea behind this PR is NOT to change `plugin` code... issue #1616 was created to discuss if we should keep the `import/no-extraneous-dependencies` disabled or explicitly define every dependency (ex: moment.js)


#### Testing instructions

1. Generate a release: `npm run build:release`
2. Verify that the generated zip file (`woocommerce-payments.zip`) contains the file `./languages/woocommerce-payments.pot`

-------------------
